### PR TITLE
change save button to fork

### LINKED ISSUE #432: Change Save Button to "Fork" when it's not your snippet
Go to the editor page for any snippet you don't own, it shouldn't say Save because you can't save, but you can fork it to make it your own!

/bounty $5

### PR DESCRIPTION
**Pull request by Anshgrover23**
/claim #432 
Fixes #432 

reference video:

https://github.com/user-attachments/assets/f86c05a6-bfd3-44c7-9967-f2ec57ce3bd8

### CODE REVIEW

File: src/components/EditorNav.tsx [hunk 1]
```diff
@@ -94,6 +96,55 @@ export default function EditorNav({
   const { toast } = useToast()
   const qc = useQueryClient()
 
+  const useForkSnippetMutation = ({
```
Comment (seveibar) [inline src/components/EditorNav.tsx:99]: create this in a separate file, hooks are never created inside render functions

**Review by seveibar — CHANGES_REQUESTED** [review]

**Review by seveibar — APPROVED** [review]
make sure tests pass before merging

**Comment by hunnywar** [comment]
@seveibar the tests are not passing from like few commits on this repo, as I am working why some of the tests are flaky, you can see this pr does not break anything related to that tests, so is this possible we can merge this pr

**Comment by Anshgrover23** [comment]
@seveibar i thinks the tests are failing from like previous merged prs and i am resolving why they are failing as given a responsibility to me by you for maintaining the health check of them but they are much flaky and this pr does not related to the test check failing problem, so is this possible we can merge this pr. As I am resolving playwright tests.

**Pull request merged**